examples/examples/valuable.rs

```diff
@@ -0,0 +1,69 @@
+#![allow(dead_code)]
+//! This example shows how a value may be recorded using valuable and as an alternative recording
+//! it without valuable but just using the debug printout. Valuable is more flexible however, as
+//! a user can implement a custom `tracing_core::Visit` and implement
+//! `tracing_core::Visit::record_value` to handle the valuable and extract the fields, whereas with
+//! the debug printout you'd need to parse the string to extract the values.
+#[cfg(tracing_unstable)]
+use tracing::field::valuable;
+use tracing::{info, info_span};
+#[cfg(tracing_unstable)]
+use valuable::Valuable;
+
+#[cfg_attr(tracing_unstable, derive(Valuable))]
+#[derive(Clone, Debug)]
+struct User {
+    name: String,
+    age: u32,
+    address: Address,
+}
+
+#[cfg_attr(tracing_unstable, derive(Valuable))]
+#[derive(Clone, Debug)]
+struct Address {
+    country: String,
+    city: String,
+    street: String,
+}
+
+#[cfg(tracing_unstable)]
+fn main() {
+    tracing_subscriber::fmt()
+        .with_max_level(tracing::Level::TRACE)
+        .init();
+
+    let user = User {
+        name: "Arwen Undomiel".to_string(),
+        age: 3000,
+        address: Address {
+            country: "Middle Earth".to_string(),
+            city: "Rivendell".to_string(),
+            street: "leafy lane".to_string(),
+        },
+    };
+
+    let span = info_span!("Processing", user = valuable(&user));
+    let _handle = span.enter();
+    info!("Nothing to do");
+}
+
+#[cfg(not(tracing_unstable))]
+fn main() {
+    tracing_subscriber::fmt()
+        .with_max_level(tracing::Level::TRACE)
+        .init();
+
+    let user = User {
+        name: "Arwen Undomiel".to_string(),
+        age: 3000,
+        address: Address {
+            country: "Middle Earth".to_string(),
+            city: "Rivendell".to_string(),
+            street: "leafy lane".to_string(),
+        },
+    };
+
+    let span = info_span!("Processing", user = ?user);
```

Comment:
would it be possible to unify this so that only the actual use of `valuable` is feature flagged? something like:
```suggestion
fn main() {
    tracing_subscriber::fmt()
        .with_max_level(tracing::Level::TRACE)
        .init();

    let user = User {
        name: "Arwen Undomiel".to_string(),
        age: 3000,
        address: Address {
            country: "Middle Earth".to_string(),
            city: "Rivendell".to_string(),
            street: "leafy lane".to_string(),
        },
    };

    // if the `valuable` feature is enabled, record `user` using its'
    // `valuable::Valuable` implementation:
    #[cfg(tracing_unstable)]
    let span = info_span!("Processing", user = valuable(&user));
    
    // otherwise, record `user` using its `fmt::Debug` implementation:
    #[cfg(not(tracing_unstable))]
    let span = info_span!("Processing", user = ?user);
    
}
```